Automattic/wp-calypso
# Overview: Apply design to new bandwidth and site views

Related to: 
- 8399-gh-Automattic/dotcom-forge
- 8400-gh-Automattic/dotcom-forge

## Proposed Changes

Styles the bandwidth usage and site view data on `/overview`, in accordance with the provided design specs. Both of these metrics are displayed for all site, but the CTA link in the bandwidth section will vary depending on the state of the site. Atomic sites get a link taking them to site monitoring, sites with an active plan that are ready to go atomic get a link to activate hosting features, and simple sites that don't have atomic-eligible plans get a link to the upgrade nudge. See design image below.

Each of these new stats will also render a placeholder when necessary (i.e. data isn't available, or is still loading). 

The tree different states of a site:
<img width="1537" alt="screen-2024-08-06-at-08 51 50" src="https://github.com/user-attachments/assets/d44ae7c1-f7e0-438f-aae8-db3b4b48a480">

Placeholders (please ignore the older copy, it's from a previous iteration of this PR):
<img width="500" alt="image" src="https://github.com/user-attachments/assets/4b17114b-31e2-4e3c-aeac-13da8564fb53">

## Why are these changes being made?

- Part of a larger project to bring relevant hosting details to our customers
- see pdKhl6-45r-p2 for full context/discussion

## Testing Instructions

To test this PR, begin by activating the local feature flag:
- Create a .env file in the root of your local calypso repo
- In that file, add `ACTIVE_FEATURE_FLAGS='hosting-overview-refinements'`
- `yarn start`

Then you can test:
- Pull up `/overview` for an active Atomic site. If necessary, visit the site a few times and wait a bit for the visits to be tracked.
- Confirm that underneath the Storage meter, you see Bandwidth data with a "Monitor site performance" link
- Confirm that link takes you to the Monitoring tab
- Back on `/overview` confirm that you also have Site Views data with a "Visit stats" link
- Confirm that link takes you to the site's stats, in Month view
- Create a new site with a Business plan. Do not activate hosting features yet. Do not visit the front end of the site 
- Pull up the new site's `/overview`
- Confirm that underneath the Storage meter, you see Bandwidth listed as "Not available" with a CTA link to activate hosting features
- Confirm that link takes you to the Hosting Features tab where you can see the proper activation button
- Back on `/overview` confirm that you also have Site Views data reporting no "No views yet for this month" and a "Visit stats" link
- Confirm that link takes you to the site's stats, in Month view
- Finally, visit `/overview` for a simple site without an atomic-eligible plan
- Confirm that bandwidth is once again "Not available" and that this time the link takes you to the upgrade nudge
